packages/SwingSet/src/vats/comms/dispatch.js

```diff
@@ -141,70 +141,59 @@ export function buildCommsDispatch(
     return sendFromKernel(target, method, args, result);
   }
 
-  function deliver(target, method, args, resultP) {
-    const result = doDeliver(target, method, args, resultP);
-    state.processMaybeFree();
-    return result;
-  }
-
-  function notify(resolutions) {
-    resolveFromKernel(resolutions);
-    state.processMaybeFree();
-  }
-
-  function dropExports(vrefs) {
-    assert(Array.isArray(vrefs));
-    vrefs.map(vref => insistVatType('object', vref));
-    vrefs.map(vref => assert(parseVatSlot(vref).allocatedByVat));
-    // console.log(`-- comms ignoring dropExports`);
-  }
-
-  function retireExports(vrefs) {
-    assert(Array.isArray(vrefs));
-    vrefs.map(vref => insistVatType('object', vref));
-    vrefs.map(vref => assert(parseVatSlot(vref).allocatedByVat));
-    // console.log(`-- comms ignoring retireExports`);
-  }
-
-  function retireImports(vrefs) {
-    assert(Array.isArray(vrefs));
-    vrefs.map(vref => insistVatType('object', vref));
-    vrefs.map(vref => assert(!parseVatSlot(vref).allocatedByVat));
-    // console.log(`-- comms ignoring retireImports`);
-  }
-
-  function dispatch(vatDeliveryObject) {
+  function doDispatch(vatDeliveryObject) {
     const [type, ...args] = vatDeliveryObject;
     switch (type) {
       case 'message': {
         const [targetSlot, msg] = args;
         insistMessage(msg);
-        deliver(targetSlot, msg.method, msg.args, msg.result);
-        return;
+        doDeliver(targetSlot, msg.method, msg.args, msg.result);
+        break;
       }
       case 'notify': {
         const [resolutions] = args;
-        notify(resolutions);
-        return;
+        resolveFromKernel(resolutions);
+        break;
       }
       case 'dropExports': {
         const [vrefs] = args;
-        dropExports(vrefs);
-        return;
+        assert(Array.isArray(vrefs));
+        vrefs.map(vref => insistVatType('object', vref));
```

Comment:
when not using the return value, `forEach` seems more appropriate than `map`

Reply:
There's lots of JS lore to the effect of "never use `forEach`", but I'm not sure the extent to which this is superstition, or a nana's pot roast thing, or whether there's some subtlety I haven't followed.